sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionalSimplification.FromCSharp8.cs

```diff
@@ -351,4 +351,28 @@ public int NestedTernary(bool condition1, bool condition2, int a, int b, int c)
             return c;
         }
     }
+
+    public int? NestedNullCoalescing(bool condition, int? a, int b)
+    {
+        if (condition) // Compliant, changing this will lead to nested conditionals
```

Comment:
Are you sure about this? Having
```
return condition
  ? b
  : a ?? b;
```
seems simple and understandable even with more complicated expressions. If I remember what Ann said about the complexity metrics than `??` is considered simple by it's purpose.

I'd prefer to ban switch expressions instead.